lib/fluent/plugin/filter_parser.rb

```diff
@@ -57,26 +57,30 @@ def configure(conf)
     FAILED_RESULT = [nil, nil].freeze # reduce allocation cost
     REPLACE_CHAR = '?'.freeze
 
-    def filter_with_time(tag, time, record)
-      raw_value = @accessor.call(record)
-      if raw_value.nil?
-        if @emit_invalid_record_to_error
-          router.emit_error_event(tag, time, record, ArgumentError.new("#{@key_name} does not exist"))
-        end
-        if @reserve_data
-          return time, handle_parsed(tag, record, time, {})
-        else
-          return FAILED_RESULT
+    def filter_stream(tag, es)
+      new_es = Fluent::MultiEventStream.new
+      es.each do |time, record|
+        begin
+          raw_value = @accessor.call(record)
+          if raw_value.nil?
+            new_es.add(time, handle_parsed(tag, record, time, {})) if @reserve_data
+            router.emit_error_event(tag,time,record,ArgumentError.new("#{@key_name} does not exist")) if @emit_invalid_record_to_error
+          else
+            filter_one_record(tag, time, record, raw_value) do |result_time, result_record|
+              new_es.add(result_time, result_record) if result_record
+            end
+          end
+        rescue => e
+          router.emit_error_event(tag, time, record, e)
```

Comment:
```suggestion
          router.emit_error_event(tag, time, record, e) if @emit_invalid_record_to_error
```

Reply:
Actually @daipom . The previous behaviour was we raise exception if emit error to event. Here also I followed the same. So your change will not have impact and changes diff structure. Isnit fine if it stays same